lambdas/scripts/bulk_upload_column_change.py

```diff
@@ -0,0 +1,185 @@
+
+        if self.found_previous_progress():
+            self.logger.info("Resuming from previous job")
+            self.resume_previous_progress()
+        else:
+            self.logger.info("Starting a new job")
+            self.initialise_new_job()
+
+        try:
+            self.run_update()
+        except Exception as e:
+            self.logger.error(e)
+            raise e
+
+    def run_update(self):
+        if len(self.progress) == 0:
+            self.logger.info(
+                f'No record found in local progress file. Please try removing the local progress file: "{self.progress}"'
+            )
+            exit()
+
+        patients_to_be_updated = [
+            doc
+            for [doc, status] in self.progress.items()
+            if not status.update_completed
+        ]
+        if not patients_to_be_updated:
+            self.logger.info(
+                "Already updated the ODS codes for all patients in previous run."
+            )
+            exit()
+
+        self.update_patient_ods()
+
+        self.logger.info("Finished updating all patient's ODS codes")
+
+    def update_patient_ods(self):
+        for doc_id, document in self.progress.items():
+            logging.info(f"Updated ODS code for {doc_id}")
+
+            updated_fields = {
+                ATTRIBUTE_TO_ADD: document.ods_code,
+                "UploaderOdsCode": UPLOADER_ODS_CODE,
+            }
+            self.dynamo_service.update_item(
+                table_name=self.table_name,
+                key=doc_id,
+                updated_fields=updated_fields,
+            )
+            logging.info(f"REMOVING {ATTRIBUTE_TO_REMOVE} for {doc_id}")
+            self.dynamo_service.remove_attribute_from_item(
+                table_name=self.table_name, key=doc_id, attribute=ATTRIBUTE_TO_REMOVE
+            )
+            self.logger.info(f"Updated ODS code for patient: {document}")
+
+            self.progress[doc_id].update_completed = True
+        self.save_progress()
+
+    def initialise_new_job(self):
+        all_entries = self.list_all_entries()
+        if len(all_entries) == 0:
+            self.logger.info(
+                f"No records was found in table {self.table_name}. Please check the table name."
+            )
+            exit()
+
+        self.progress = self.build_progress_dict(all_entries)
+
+    def resume_previous_progress(self):
+        try:
+            with open(self.progress_store, "r") as f:
+                json_str = f.read()
+                self.progress = TypeAdapter(Dict[str, ProgressForDoc]).validate_json(
+                    json_str
+                )
+        except FileNotFoundError:
+            self.logger.info("Cannot find a progress file. Will start a new job.")
+            self.initialise_new_job()
+
+    def found_previous_progress(self) -> bool:
+        return os.path.isfile(self.progress_store)
+
+    def save_progress(self):
+        with open(self.progress_store, "wb") as f:
```

Comment:
Binary mode 'wb' is generally for writing binary data, such as images or files that contain non-text data. 'w' mode only needed for json